libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.cpp

```diff
@@ -87,11 +87,16 @@ AP_BattMonitor_UAVCAN* AP_BattMonitor_UAVCAN::get_uavcan_backend(AP_UAVCAN* ap_u
 void AP_BattMonitor_UAVCAN::handle_battery_info(const BattInfoCb &cb)
 {
     WITH_SEMAPHORE(_sem_battmon);
-    _interim_state.temperature = cb.msg->temperature;
     _interim_state.voltage = cb.msg->voltage;
     _interim_state.current_amps = cb.msg->current;
     _soc = cb.msg->state_of_charge_pct;
 
+    if (!isnanf(cb.msg->temperature) && cb.msg->temperature > 0) {
```

Comment:
We tend to use `is_positive` rather than `> 0` for floats (it uses an epsilon).  OTOH, if this is a `float16` the epsilon is probably wrong :-)

Reply:
Very good point, I'll keep this in mind for future 🙂